java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java

```diff
@@ -343,13 +337,20 @@ public void visitTypeCast(TypeCastTree tree) {
 
     private void checkDeferredConstraint() {
       SymbolicValue sv = programState.peekValue();
-      if (sv instanceof DeferredStatusHolderSV) {
-        addZeroConstraint(sv, ((DeferredStatusHolderSV) sv).deferredStatus);
+      if (sv instanceof DeferredConstraintHolderSV) {
+        addZeroConstraint(sv, ((DeferredConstraintHolderSV) sv).deferredConstraint);
       }
     }
 
-    private void addZeroConstraint(SymbolicValue sv, DivByZeroStatus status) {
-      programState = programState.addConstraint(sv, new ZeroConstraint(status));
+    private void addZeroConstraint(SymbolicValue sv, ZeroConstraint zeroConstraint) {
+      if(zeroConstraint == null) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "false" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2583)